File: repos_estudiantes_ALVAREZ_primera_entrega/ARUQUIPA_VARGAS_ANA_GLORIA/POLIMORFISMO/Ejercicio3py_java/Main.java
package Ejercicio3py_java;
public class Main {
    public static void main(String[] args) {
        Matriz m1 = new Matriz();
        System.out.println("Matriz identidad:");
        m1.mostrar();

        float[][] datos = new float[10][10];
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                datos[i][j] = i + j;
            }
        }

        Matriz m2 = new Matriz(datos);
        System.out.println("\nMatriz personalizada:");
        m2.mostrar();

        Matriz suma = m1.sumar(m2);
        System.out.println("\nSuma de m1 + m2:");
        suma.mostrar();

        Matriz resta = m2.restar(m1);
        System.out.println("\nResta de m2 - m1:");
        resta.mostrar();

        System.out.println("\n¿m1 es igual a m2?: " + m1.igual(m2));
        System.out.println("¿m1 es igual a otra identidad?: " + m1.igual(new Matriz()));
    }
}

File: repos_estudiantes_ALVAREZ_primera_entrega/ARUQUIPA_VARGAS_ANA_GLORIA/POLIMORFISMO/Ejercicio3py_java/main.py
from Matriz import Matriz
if __name__ == "__main__":
    m1 = Matriz()
    print("Matriz identidad:")
    m1.mostrar()

    datos = [[float(i + j) for j in range(10)] for i in range(10)]
    m2 = Matriz(datos)
    print("\nMatriz personalizada:")
    m2.mostrar()

    suma = m1.sumar(m2)
    print("\nSuma:")
    suma.mostrar()

    resta = m2.restar(m1)
    print("\nResta:")
    resta.mostrar()

    print("\n¿m1 igual a m2?:", m1.igual(m2))
    print("¿m1 igual a m1?:", m1.igual(Matriz())) 
